### PR TITLE
Add flag to drop metrics that don't match the mapping

### PR DESCRIPTION
**Pull request by victorcete**
Hello there!

First of all, thanks for your work on this, it is a very useful tool :+1: 

We have a very large data set of metrics and we are trying to forward small subsets of metrics onto a different processing/analytics storage, but we don't want all the metrics. We only want those metrics that matches our mapping file configuration.

I added just a bunch of lines to enable that behaviour.

Cheers!

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ var (
 	graphiteAddress  = flag.String("graphite.listen-address", ":9109", "TCP and UDP address on which to accept samples.")
 	mappingConfig    = flag.String("graphite.mapping-config", "", "Metric mapping configuration file name.")
 	sampleExpiry     = flag.Duration("graphite.sample-expiry", 5*time.Minute, "How long a sample is valid for.")
+	strictMatch      = flag.Bool("graphite.strict-match", false, "Only store metrics that match the mapping configuration.")
```
Comment (brian-brazil) [inline main.go]: mapping-strict-match maybe so it's closer to mapping-config?

File: main.go [hunk 2]
```diff
@@ -96,6 +97,10 @@ func (c *graphiteCollector) processLine(line string) {
 		name = labels["name"]
 		delete(labels, "name")
 	} else {
+		// if graphite.strict-match flag is set, we will drop this metric
```
Comment (brian-brazil) [inline main.go]: Comments should be full sentances, so:

If graphite.strict-match flag is set, we will drop this metric.

**Comment by brian-brazil** [comment]
In case you're not aware you can also filter these at the Prometheus level (http://www.robustperception.io/dropping-metrics-at-scrape-time-with-prometheus/) though it's best if you can not send them at the source.

File: main.go (hunk 1 above)
Comment (victorcete) [inline main.go]: sure, makes total sense

File: main.go (hunk 2 above)
Comment (victorcete) [inline main.go]: noted, thanks!

**Comment by victorcete** [comment]
Thanks for the info, I didn't know that feature but as you said, I also prefer to avoid sending unnecessary metrics.

**Comment by brian-brazil** [comment]
Looks good, can you squash the commits please?

**Comment by victorcete** [comment]
Sure, I wasn't familiar with that. I'll update it, thanks for the suggestion

**Comment by victorcete** [comment]
Done :) hopefully everything is now correct. Thanks!

**Pull request merged**